mops-torch/src/opsa.cpp

```diff
@@ -45,9 +45,25 @@ torch::Tensor OuterProductScatterAdd::forward(
             );
         });
     } else {
-        C10_THROW_ERROR(
-            ValueError, "outer_product_scatter_add is not implemented for device " + A.device().str()
+
```

Comment:
Please leave a catch-all `else` here, and wrap the code below in `else if (device == cuda)`

Reply:
fixed!